### ISSUE TITLE
SimpleRNN layer is partially parsed by hls4ml (weights are generated), but not instantiated in the generated HLS C++ (myproject.cpp

### ISSUE THREAD

**Issue by jachoulin304** [fastmachinelearning/hls4ml#1464]
## Prerequisites
Please make sure to check off these prerequisites before submitting a bug report.
- [V ] Test that the bug appears on the current version of the master branch. Make sure to include the commit hash of the commit you checked out.
- [V ] Check that the issue hasn't already been reported, by checking the currently open issues.
- [V ] If there are steps to reproduce the problem, make sure to write them down below.
- [V ] If relevant, please include the hls4ml project files, which were created directly before and/or after the bug.

## Quick summary
SimpleRNN layer is partially parsed by hls4ml (weights are generated), but not instantiated in the generated HLS C++ (myproject.cpp), leading to a broken computation graph and HLS synthesis failure.

## Details
Steps to Reproduce
1. Install environment:
pip install hls4ml tensorflow==2.13

2. Create a minimal Keras model:
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Embedding, SimpleRNN, Dense

model = Sequential([
    Embedding(input_dim=1000, output_dim=4, input_length=16, name="embed"),
    SimpleRNN(16, name="rnn1"),
    Dense(1, activation="sigmoid", name="output")
])

3. Convert using hls4ml:
import hls4ml

config = hls4ml.utils.config_from_keras_model(model, granularity="name")

hls_model = hls4ml.converters.convert_from_keras_model(
    model,
    hls_config=config,
    io_type="io_stream",
    output_dir="hls4ml_prj",
    part="xczu7ev-ffvc1156-2-e"
)

4. Generate project:
hls_model.build(csim=False, synth=False)

### Expected behavior
The generated HLS code should include a recurrent layer call, e.g.:
**nnet::simple_rnn(...);**
and the computation graph should be:
Embedding → SimpleRNN → Dense

### Actual behavior
myproject.cpp does not contain any SimpleRNN / recurrent layer call
However, weights for the RNN layer are generated and loaded, e.g.:
nnet::load_weights_from_txt<rnn1_weight_t, ...>(...)
nnet::load_weights_from_txt<rnn1_recurrent_weight_t, ...>(...)
The generated computation graph becomes:
Embedding → Dense
Intermediate buffer (e.g., layer3_out) has:
no producer
but is consumed by Dense

This leads to HLS errors such as:
ERROR: Internal stream 'layer3_out' failed dataflow checking

**Comment by kr1xh** [comment]
Hi! I'd like to work on this issue.

From the reproduction, it looks like the SimpleRNN layer is being parsed far enough for its weights to be generated, but the recurrent operation itself is missing from the generated HLS graph/code.

I'd like to trace the SimpleRNN path from the Keras converter through the model graph and backend code generation, identify where the layer is being dropped, and add a regression test for the Embedding → SimpleRNN → Dense case.

If there are no objections, I'd be happy to investigate this.

**Comment by vloncar** [comment]
By all means, please do. Thanks.

BTW, if you plan on using agents for this, we would appreciate it if you could try #1527 and follow the guidelines of #1514. We welcome feedback on the agent skills.
